### PR TITLE
first draft

### PR DESCRIPTION
**Pull request by knaaptime**
here's a first draft of the recipe for quilt3. I bootstrapped the dependencies from your `setup.py` (i checked, all of them are available on conda-forge) but some of the more complex specifications (e.g. `xattr >=0.9.6; platform_system!="Windows"'`) might need to be reformatted, I'm not sure. The quickest way to know for sure might be to just open this PR in the feedstock... the conda-forge devs are really responsive

### CODE REVIEW

File: recipes/quilt/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+{% set name = "quilt3" %}
+{% set version = "3.0.5" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/quiltdata/{{ name }}/archive/{{ version }}.tar.gz
```
Comment (akarve) [inline recipes/quilt/meta.yaml:9]:
```suggestion
  url: https://github.com/quiltdata/{{ repo }}/archive/{{ version }}.tar.gz
```

File: recipes/quilt/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+{% set name = "quilt3" %}
+{% set version = "3.0.5" %}
+
```
Comment (akarve) [inline recipes/quilt/meta.yaml:3]:
```suggestion
{% repo = "quilt" %}

```
`quilt3` is indeed the right package name, but it lives under `quilt` on GitHub.

File: recipes/quilt/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/quiltdata/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: eec5bee2d0037df8b726b88fba373596e3fc7a15f473013faeef20a5cb0415b1
```
Comment (akarve) [inline recipes/quilt/meta.yaml]: Did I do this wrong?
```
(glam) apk-mbp-3:t4 karve$ shasum -a 256 ~/Downloads/quilt-3.0.5.tar.gz 
ded69d039949de8293b0c98e95cd0a958eb7e68c93106b1f35254dd4691685c4
```

File: recipes/quilt/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  license_family: Apache
+  license_file: LICENSE
+  summary: 'Quilt: where data comes together'
+
+  description: |
+    Quilt is a data management tool designed for data discoverability, data dependency
+    management, and data version control using `data packages <https://blog.quiltdata.com/data-packages-for-fast-reproducible-python-analysis-c74b78015c7f>`_.
+    The `quilt` PyPi package allows you to build, push, and pull data packages in Quilt using Python.
+    Visit the `documentation quickstart <https://quiltdocs.gitbook.io/quilt/quickstart>`_ for more information.
+  dev_url: https://https://github.com/quiltdata/quilt
```
Comment (akarve) [inline recipes/quilt/meta.yaml]:
```suggestion
  dev_url: https://github.com/quiltdata/quilt
```

File: recipes/quilt/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+about:
+  home: https://quiltdata.com/
+  license: Apache 2.0
+  license_family: Apache
+  license_file: LICENSE
+  summary: 'Quilt: where data comes together'
+
+  description: |
+    Quilt is a data management tool designed for data discoverability, data dependency
```
Comment (akarve) [inline recipes/quilt/meta.yaml]:
```suggestion
    Quilt is infrastructure for data-driven teams to store, version, deploy and iterate on models and data.
    Data in Quilt are visual, discoverable, and reproducible.
```

File: recipes/quilt/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    management, and data version control using `data packages <https://blog.quiltdata.com/data-packages-for-fast-reproducible-python-analysis-c74b78015c7f>`_.
+    The `quilt` PyPi package allows you to build, push, and pull data packages in Quilt using Python.
+    Visit the `documentation quickstart <https://quiltdocs.gitbook.io/quilt/quickstart>`_ for more information.
```
Comment (akarve) [inline recipes/quilt/meta.yaml]: OK to drop above 3 lines

**Review by akarve — COMMENTED** [review]
Thanks for taking the first cut @knaaptime. A few comments inline.

File: recipes/quilt/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  sha256: eec5bee2d0037df8b726b88fba373596e3fc7a15f473013faeef20a5cb0415b1
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+
+requirements:
+  host:
+    - python >=3
```
Comment (akarve) [inline recipes/quilt/meta.yaml]: I think we need `>=3.6`?

File: recipes/quilt/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    - pip
+  run:
+    - python >=3
+    - appdirs>=1.4.0
+    - aws-requests-auth >=0.4.2
+    - boto3 >=1.8.0
```
Comment (akarve) [inline recipes/quilt/meta.yaml:27]:
```suggestion
    - boto3>=1.8.0 
```

File: recipes/quilt/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    - python >=3
+    - appdirs>=1.4.0
+    - aws-requests-auth >=0.4.2
+    - boto3 >=1.8.0 
+    - elasticsearch==6.3.1
+    - jsonlines==1.2.0
+    - numpy>=1.14.0                   # required by pandas, but missing from its dependencies.
+    - packaging>=16.8
+    - pandas>=0.19.2
+    - pyarrow ==0.13              # as of 7/5/19: linux/circleci bugs on 0.14
```
Comment (akarve) [inline recipes/quilt/meta.yaml]:
```suggestion
    - pyarrow==0.13              # as of 7/5/19: linux/circleci bugs on 0.14
```

File: recipes/quilt/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    - requests>=2.12.4
+    - ruamel.yaml<=0.15.70
+    - tqdm>=4.26.0
+    - urllib3==1.24          # required by requests
+    - xattr>=0.9.6
+    - humanize
+
+test:
+  imports:
```
Comment (akarve) [inline recipes/quilt/meta.yaml]: i'm checking to see how to run [our standard CircleCI tests here](https://github.com/quiltdata/quilt/blob/master/.circleci/config.yml#L33)

File: recipes/quilt/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    - quilt3
+
```
Comment (akarve) [inline recipes/quilt/meta.yaml]:
```suggestion
  requires:
    - pytest
  source_files:
    - api/python/tests
  commands:
    - cd api/python
    - pytest tests
```

Comment (akarve) [inline recipes/quilt/meta.yaml:43]:
```suggestion
  requires:
    - pytest
  source_files:
    - api/python/tests
  commands:
    - cd api/python
    - pytest tests

```

File: recipes/quilt/meta.yaml (hunk 3 above)
Comment (knaaptime) [reply recipes/quilt/meta.yaml]: no, you're right, not sure where i got that other hash

File: recipes/quilt/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/quiltdata/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: ded69d039949de8293b0c98e95cd0a958eb7e68c93106b1f35254dd4691685c4
+
+build:
```
Comment (akarve) [inline recipes/quilt/meta.yaml:12]:
```suggestion
build:
  skip: true  # [win]
```

File: recipes/quilt/meta.yaml [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    - jsonlines==1.2.0
+    - numpy>=1.14.0                   # required by pandas, but missing from its dependencies.
+    - packaging>=16.8
+    - pandas>=0.19.2
+    - pyarrow ==0.13              # as of 7/5/19: linux/circleci bugs on 0.14
+    - requests>=2.12.4
+    - ruamel.yaml<=0.15.70
+    - tqdm>=4.26.0
+    - urllib3==1.24          # required by requests
+    - xattr>=0.9.6
```
Comment (akarve) [inline recipes/quilt/meta.yaml]:
```suggestion
    - xattr>=0.9.6 # [not win]
```
If this works you can possibly restore Windows builds

**Comment by drernie** [comment]
Merging this in, as it was approved.  Can decide later if we want. to push upstream.

**Pull request merged**